src/conversions/sample.rs

```diff
@@ -178,3 +177,76 @@ impl Sample for f32 {
         0.0
     }
 }
+
```

Comment:
Were the tests removed? These can be shorter but actually useful to check for edge cases. Type conversions' behavior is not always obvious.

Reply:
Is this comment about the tests or the calculation change?
Another option to avoid overflows would be to use smaller integer types for the coefficient,  I think `u32` is too much for the use case. Or, maybe use a float as coefficient.

Reply:
How one would test the performance here? Setting up a lot of conversions in a row?